engine/src/main/java/org/destinationsol/game/SolContactListener.java

```diff
@@ -50,24 +50,28 @@ public void beginContact(Contact contact) {
             EntityRef entityB = (EntityRef) dataB;
             entitySystemManager.sendEvent(new ContactEvent(entityB, contact), entityA);
             entitySystemManager.sendEvent(new ContactEvent(entityA, contact), entityB);
-            return;
         }
 
-        if (dataA instanceof SolObject && dataB instanceof SolObject) {
-            SolObject oA = (SolObject) contact.getFixtureA().getBody().getUserData();
-            SolObject oB = (SolObject) contact.getFixtureB().getBody().getUserData();
-
-            boolean aIsProj = oA instanceof Projectile;
-            if (!aIsProj && !(oB instanceof Projectile)) {
-                return;
-            }
-
-            Projectile proj = (Projectile) (aIsProj ? oA : oB);
-            SolObject o = aIsProj ? oB : oA;
-            proj.setObstacle(o, myGame);
+        //TODO This is a patch to smooth over contact between an Entity and a Projectile. Once Projectile has been converted
+        // to be an Entity, this can be removed.
+        if (dataA instanceof EntityRef) {
+            dataA = new SolObjectEntityWrapper((EntityRef) dataA);
+        }
+        if (dataB instanceof EntityRef) {
+            dataB = new SolObjectEntityWrapper((EntityRef) dataB);
         }
 
-        //TODO handle contact between an entity and a SolObject
+        //TODO This is legacy code for handling contact with a Projectile, which currently is designed to work with SolObjects.
+        // Once Projectile has been converted to be an entity, this should be refactored.
+        SolObject oA = (SolObject) dataA;
+        SolObject oB = (SolObject) dataB;
```

Comment:
Again here, you removed an earlier check to make sure that `dataA` and `dataB` are both `SolObjects`. It was probably there for a good reason :)

Reply:
I put that in [here](https://github.com/MovingBlocks/DestinationSol/pull/543/files#diff-36cb9a67d69750ed5e3e1e539965b7c1R56). Before I set up the contact handling with `Projectiles`, I needed to make sure that the game wouldn't crash. This removes the need for the check, because it now works properly.